### ISSUE TITLE
Better Tooling ?

### ISSUE THREAD

**Issue by IceWreck** [packit/dashboard#87]
Our current way to compile/transpile Javascript is a [custom webpack config](https://github.com/packit/dashboard/blob/main/webpack.config.js), which we needed because React's default preconfigured Create React App toolchain (which itself is based on webpack) couldn't inject environment variables in JavaScript at build time.

I realized that it can do that now, but only with env vars starting with `REACT_APP_`.

React generally recommends writing your own webpack config if using it with something like flask, but we don't need any advanced webpack features so I think its time to retire our config.

On the flip side, switching to Create React App would give us:

- Proper Code splitting and our final prod js bundle size is halved. (tested the half thing by compiling dashboard with CRA, but I think it can be reduced even more if we make it aware of routes later someday)
- No need to maintain or rely on [webpack.config.js](https://github.com/packit/dashboard/blob/main/webpack.config.js)
- Warnings about bad JS/React practices
- Better debugging (and better error messages during dev)

All this can be done by writing your own config, but its too complicated and I'm not a fan of the JS/npm ecosystem.

BTW, if we do this, we'll have to provide another env variable in openshift, something like `REACT_APP_DEPLOYMENT` because it needs to start with that.

All of the code will remain the same, but there might be some changes to the folder structure. I can do it next weekend if everyone is okay with it.

**Comment by TomasTomecek** [comment]
> I'm not a fan of the JS/npm ecosystem.

me too :)

+1 on simplifying things and dropping the config if we don't need it any more

**Comment by IceWreck** [comment]
How do you provide an environment variable at build time ? There is no ARG command in the Dockerfile currently. Is there another way to do it that I'm unaware of ?

**Comment by mfocko** [comment]
> How do you provide an environment variable (like `DEPLOYMENT`) at build time ? There is no ARG command in the Dockerfile currently. Is there another way to do it that I'm unaware of ?

Example here: https://github.com/packit/deployment/blob/main/openshift/packit-service.yml.j2#L51
And this one is for dashboard: https://github.com/packit/deployment/blob/main/openshift/dashboard.yml.j2#L48

**Comment by IceWreck** [comment]
> > How do you provide an environment variable (like `DEPLOYMENT`) at build time ? There is no ARG command in the Dockerfile currently. Is there another way to do it that I'm unaware of ?
> 
> Example here: https://github.com/packit/deployment/blob/main/openshift/packit-service.yml.j2#L51
> And this one is for dashboard: https://github.com/packit/deployment/blob/main/openshift/dashboard.yml.j2#L48

Thanks !

![image](https://user-images.githubusercontent.com/18102790/114778948-1c91c280-9d93-11eb-8b01-30d4466c3e8c.png)

These look like they're provided at runtime, after the image is built. (please correct me if I'm wrong though) That is okay currently, but with the new tooling we will have to pass it at build time, with a build arg providing the env variable, with different env vars for the stg and prod image tags.

Example:

```
podman build --rm \
    --build-arg REACT_APP_API_URL=https://stg.packit.dev/api \
    -t quay.io/packit/dashboard:stg -f Dockerfile .
```

**Comment by mfocko** [comment]
oh ok, sorry
what about this then? https://github.com/packit/dashboard/blob/main/.github/workflows/build-and-push.yml#L41
looks like it is possible with that action: https://github.com/marketplace/actions/buildah-build#inputs-for-build-without-dockerfile

**Comment by IceWreck** [comment]
> what about this then? https://github.com/packit/dashboard/blob/main/.github/workflows/build-and-push.yml#L41
> looks like it is possible with that action: https://github.com/marketplace/actions/buildah-build#inputs-for-build-without-dockerfile

Yes, thanks. This is it. We can specify build args here.

**Comment by mfocko** [comment]
Fixed by #96?

@IceWreck or is there something more to do?

**Comment by IceWreck** [comment]
No, its done
